src/test/regress/sql/multi_complex_expressions.sql

```diff
@@ -52,3 +52,21 @@ SELECT count(*) FROM lineitem JOIN orders ON l_orderkey = o_orderkey
 -- Check that we make sure local joins are between columns only.
 
 SELECT count(*) FROM lineitem, orders WHERE l_orderkey + 1 = o_orderkey;
+
+-- Check that we can use UDFs when using repartitioned subqueries
+
+DROP FUNCTION IF EXISTS median(double precision[]);
+
+CREATE FUNCTION median(double precision[]) RETURNS double precision 
+LANGUAGE sql IMMUTABLE AS $_$ 
+	SELECT AVG(val) FROM 
+	  (SELECT val FROM unnest($1) val 
+	   ORDER BY 1 LIMIT  2 - MOD(array_upper($1, 1), 2) 
+	   OFFSET CEIL(array_upper($1, 1) / 2.0) - 1) sub; 
+$_$;
+
+EXPLAIN SELECT * FROM (
+    	SELECT
+      	  median(ARRAY[1,2,sum(l_suppkey)]) as median, count(*)
+    	  FROM lineitem GROUP BY l_partkey) AS a
+	WHERE median > 2;
```

Comment:
verified that fix works for the original bug description. However this regression test does not verify that since the declared function is local to master only.

please see regression test ``multi_index_statements.sql`` on connecting to worker nodes within test.

You should
1 - create the function in master as you did here
2 - connect to worker1 and create the function
3 - connect to worker2 and create the function
4 - run the query and verify it works.

You would see that this test will fail if you remove the fix you put in ``worker_merge_protocol.c``